src/extension/wps/wps-core/src/main/java/org/geoserver/wps/gs/ImportProcess.java

```diff
@@ -247,6 +254,21 @@ public String execute(
         return null;
     }
 
+    private static void checkWriteAccess(WorkspaceInfo workspace) {
+        // This is essentially the same code to check the access level as used by
+        // org.geoserver.security.SecureCatalogImpl.getDataStoreByName(String, String)
+        WrapperPolicy policy =
+                GeoServerExtensions.bean(SecureCatalogImpl.class)
+                        .buildWrapperPolicy(
+                                SecurityContextHolder.getContext().getAuthentication(),
+                                workspace,
+                                workspace.getName(),
+                                MixedModeBehavior.CHALLENGE);
+        if (policy.level != AccessLevel.READ_WRITE) {
```

Comment:
As per our meeting today, moving the checks to the SecureCatalogImpl would be architecturally better, but would open a can of works vs [the REST security](https://docs.geoserver.org/maintain/en/user/security/rest.html).

So ok to do it here, but it should check whether the current user has admin access on the current workspace, something like ``((WorkspaceAccessLimits) policy.getAccessLimits()).isAdminable()``.
Which is probably similar, in practice, to what you're doing, but clearer and we hope more future proof.

Reply:
I updated it to check for admin access instead of write access.